### ISSUE TITLE
Ribbon Relative Mode doesn't reflect UI, MIDI changes

### ISSUE THREAD

**Issue by MatthiasSeeber** [nonlinear-labs-dev/C15#2696]
observed in [KW 21-22](https://github.com/nonlinear-labs-dev/C15/releases/tag/21-22):

The Ribbon Relative mode only works reliably when exclusively using the Ribbon itself. Changes to the Ribbon (from UI: inc/dec/etc or MIDI) are not propagated to the PlayController, so the changed value isn't reflected when touching the relative Ribbon, causing an unwanted jump to the previously touched position.

Proposal:
- consider/streamline PlayController Event flow (PlayGround, BBB (?), Synth) in general
- implement bidirectional Ribbon Event flow (UI, MIDI --> PlayController), solving the relative mode

**Comment by KlausStrohhaecker** [comment]
The functions are available already but don't seem to be used.
See ID's PLAYCONTROLLER_SETTING_ID_UPPER_RIBBON_VALUE amd PLAYCONTROLLER_SETTING_ID_LOWER_RIBBON_VALUE in /projects/shared/playcontroller/playcontroller-defs.h

Personally, I'd ditch the whole special features and calculations in the playcontroller for ribbon relative mode completely, That's high-level stuff the playcontroller should not have to deal with. It should just sends touch values, that's it.

**Comment by MatthiasSeeber** [comment]
But then, the PlayController would send all HW Sources except Ribbons in TCD Format [0...16000] and the Ribbons in another relative fashion. This seems to introduce many new complications in my opinion.

**Comment by KlausStrohhaecker** [comment]
No, the ribbon values are just absolute values like in the normal case. The relative mode already is working with the TDC values in seperate small function which could easily moved to somewhere else.
The whole relative mode things actually just creates overhead/detours in the playcontroller. It's controlled and parametrized by the Playground which could as well do the same for that simple function when it were moved somewhere else.

The whole idea of doing high level stuff (including return behaviour and such) on the LPC only has historic reasons when the AE and PG were on different platform that could not communicate, so the LPC was misused as a poxy.

**Comment by KlausStrohhaecker** [comment]
see /projects/playcontroller/firmware/src/main/src/tcd/nl_tcd_adc_work.c, line 300,

There also is another bug in relative mode for play control (not as edit control), the starting point is not updated/sent to the playcontroller on preset change etc.

**Comment by MatthiasSeeber** [comment]
Yes, any UI or MIDI change (except for actually touching the Ribbon, is also mentioned in #1795).

Regarding the relative mode: I suspected that the Touch Event on a Ribbon is evaluated in the LPC directly, but that doesn't seem to be the case I guess?

**Comment by KlausStrohhaecker** [comment]
Touch/release events are only known to the LPC, and a touch event additionally enables sending data until release.

**Comment by KlausStrohhaecker** [comment]
After checking the code I would think the fix (for play mode) is simple:
- when the PG enters relative mode for a ribbon, the playcontroller is notified already to do the same.
- insert the corresponding PLAYCONTROLLER_SETTING_ID_xxx_RIBBON_VALUE with the TCD start value **after** the command for relative mode (via PLAYCONTROLLER_SETTING_ID_PLAY_MODE_xxx_RIBBON_BEHAVIOUR)

For relative mode in parameter edit mode, we'll have to check...

**Comment by KlausStrohhaecker** [comment]
- [x] playcontroller relative value handling in PLAY mode

- [ ] playcontroller relative value handling in EDIT mode
PG must tell playcontroller the start value for EDIT mode as well (need a new Setting Msg and playcontroller function for this)
